### DISCUSSION TITLE
Server requirements for multiple users (100+, 200+, 500+)?

### DISCUSSION THREAD

**Discussion by FEOTB** [serenity-is/Serenity#6845] [Help]
Does anyone know what sort of hardware requirements are required for supporting multiple users?  I've been given a range of somewhere between 100 - 500 users that can access various parts of our system at any given time.  Our expected infrastructure is to have one application server and one database server that is accessible via our internal subnet.

I've been trying to find some sort of documentation on performance requirements, but I haven't been able to find anything.

Thank you very much for your time.

**Comment by volkanceylan** [comment]
There is no such magic formula for calculating number of users your server can support concurrently as it depends on your server and database resources, average number of records in frequently accessed tables, patterns of users in an average session, index structure in DB etc. Serenity adds very little overhead and can be ignored when calculating metrics comparee to DB access, locks, indexes, bandwidth etc. Anyway, I can say with an app that has hundreds of thousands of users (students), an average server can handle tens of thousands of concurrent users during peak times like registration. But it depends entirely on your case, design etc.
